### PR TITLE
chore(electric): Apply own review feedback from the Proxy PR and other fixes

### PR DESCRIPTION
**Pull request by alco**
This applies the feedback that was missed in https://github.com/electric-sql/electric/pull/421. I've also made two more cleanups:
* got rid of the need to `create extension uuid-ossp` by using the builtin `gen_random_uuid()` function
* consistently applied the mapping from PG's `NULL` to Elixir's `nil` in all epgsql connections

### CODE REVIEW

File: e2e/prisma_example/migrations-1/prisma/schema.prisma [hunk 1]
```diff
@@ -14,13 +14,13 @@ datasource db {
 // https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#baseline-your-production-environment
 
 model entries {
-  id        String  @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
+  id        String  @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
```
Comment (magnetised) [inline e2e/prisma_example/migrations-1/prisma/schema.prisma:17]: how does this play with your disallow-defaults pr?

**Review by magnetised — APPROVED** [review]
cool. didn't notice the comments re epgsql or the uuid stuff tbh.

File: e2e/prisma_example/migrations-1/prisma/schema.prisma (hunk 1 above)
Comment (alco) [reply e2e/prisma_example/migrations-1/prisma/schema.prisma:17]: This PR is actually based on the branch that rejects tables with DEFAULT clauses. As you can see, all tests are passing. These `entries` and `owned_entries` tables aren't electrified, at least not as part of the migrations in the `migrations-1/prisma` directory. I didn't dive in to investigate this nuance, to be honest.

File: components/electric/lib/electric/replication/postgres_manager.ex [hunk 2]
```diff
@@ -134,9 +134,11 @@ defmodule Electric.Replication.PostgresConnectorMng do
         {:DOWN, ref, :process, pid, reason},
         %State{pg_connector_sup_monitor: ref} = state
       ) do
-    Logger.warning(
```
Comment (davidmartos96) [inline components/electric/lib/electric/replication/postgres_manager.ex:137]: @alco Minor typo I have noticed by chance when overlooking the code :sweat_smile: 
"shutdwon"

**Pull request merged**